drv/stm32xx-sys/src/main.rs

```diff
@@ -255,20 +395,241 @@ impl idl::InOrderSysImpl for ServerImpl<'_> {
     ) -> Result<[u32; 3], RequestError<core::convert::Infallible>> {
         Ok(drv_stm32xx_uid::read_uid())
     }
+
+    fn gpio_irq_control(
+        &mut self,
+        rm: &RecvMessage,
+        disable_mask: u32,
+        enable_mask: u32,
+    ) -> Result<(), RequestError<core::convert::Infallible>> {
+        // We want to only include code for this if exti is requested.
+        // Unfortunately the _operation_ is available unconditionally, but we'll
+        // fault any clients who call it if it's unsupported (below).
+        cfg_if! {
+            if #[cfg(feature = "exti")] {
+
+                for (i, entry) in generated::EXTI_DISPATCH_TABLE.iter().enumerate() {
+                    // Only use populated rows in the table
+                    if let Some((_, tid, mask)) = entry {
+                        // Ignore anything assigned to another task
+                        if tid.index() == rm.sender.index() {
+                            // Apply disable first so that including the same
+                            // thing in both masks is a no-op.
+                            if mask & disable_mask != 0 {
+                                // Disable this source by _clearing_ the
+                                // corresponding mask bit.
+                                self.exti.cpuimr1.modify(|r, w| {
+                                    let new_value = r.bits() & !(1 << i);
+                                    // Safety: not actually unsafe, PAC didn't
+                                    // model this field right
+                                    unsafe {
+                                        w.bits(new_value)
+                                    }
+                                });
+                            }
+
+                            if mask & enable_mask != 0 {
+                                // Enable this source by _setting_ the
+                                // corresponding mask bit.
+                                self.exti.cpuimr1.modify(|r, w| {
+                                    let new_value = r.bits() | (1 << i);
+                                    // Safety: not actually unsafe, PAC didn't
+                                    // model this field right
+                                    unsafe {
+                                        w.bits(new_value)
+                                    }
+                                });
+                            }
+                        }
+                    }
+                }
+                Ok(())
```

Comment:
do you think it makes sense to punish clients who call `gpio_irq_control` with a mask that doesn't include any notifications mapped to EXTI interrupts with a reply fault? 

we could have a flag that get set if we actually touched something, and reply fault if we didn't...it seems like a potential programmer error to call this IPC with a mask that doesn't include EXTI notifications.

Reply:
In general, reply-faulting in cases like that makes it easier for people to find programming errors faster, rather than hunting around for their missed events. So, yeah, I think it probably should. I admit I considered this when writing the implementation and then decided I was in a hurry.

I've gotten clarified priorities on this and am no longer in a hurry, so, I will add this.